File: software/examples/perl/example_interrupt.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletIO16V2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your IO-16 Bricklet 2.0

# Callback subroutine for input value callback
sub cb_input_value
{
    my ($channel, $changed, $value) = @_;

    print "Channel: $channel\n";
    print "Changed: $changed\n";
    print "Value: $value\n";
    print "\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $io = Tinkerforge::BrickletIO16V2->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register input value callback to subroutine cb_input_value
$io->register_callback($io->CALLBACK_INPUT_VALUE, 'cb_input_value');

# Set period for input value (channel 4 [A4]) callback to 0.5s (500ms)
$io->set_input_value_callback_configuration(4, 500, 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/javascript/ExampleInterrupt.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your IO-16 Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var io = new Tinkerforge.BrickletIO16V2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Set period for input value (channel 4 [A4]) callback to 0.5s (500ms)
        io.setInputValueCallbackConfiguration(4, 500, false);
    }
);

// Register input value callback
io.on(Tinkerforge.BrickletIO16V2.CALLBACK_INPUT_VALUE,
    // Callback function for input value callback
    function (channel, changed, value) {
        console.log('Channel: ' + channel);
        console.log('Changed: ' + changed);
        console.log('Value: ' + value);
        console.log();
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
